yo/app/scripts/directives/filter.js

```diff
@@ -74,10 +74,21 @@ angular.module('dmpApp')
             return true;
         };
 
+        $scope.filterTypes = [
+            {id: 'NUMERIC', name: 'numeric filter'},
+            {id: 'REGEXP', name: 'regular expression'},
+            {id: 'EQUALS', name: 'equals'},
+            {id: 'NOTEQUALS', name: 'not-equals'}
+        ];
+
+        $scope.defaultFilterType = $scope.filterTypes[1];
+
         $scope.addFilter = function() {
 
+            var filter = schemaParser.fromDomainSchema($scope.dataSchema, true, true, $scope.filterTypes, $scope.defaultFilterType);
```

Comment:
The way is see it, `defaultFilterType` is not used in the view, so it needn't be defined on `$scope`.
Also, the filterTypes and the defaultType are directly related to each other, so maybe pass an object of `{types, default}` as one argument to `fromDomainSchema` and there, we only need one if-argument-is-defined check.